internal/usbgadget/hid_keyboard.go

```diff
@@ -115,19 +122,25 @@ func (u *UsbGadget) updateKeyboardState(state byte) {
 	u.keyboardStateLock.Lock()
 	defer u.keyboardStateLock.Unlock()
 
+	context := u.getHidKeyboardLoggingContext().Uint8("state", state)
+
 	if state&^ValidKeyboardLedMasks != 0 {
-		u.log.Warn().Uint8("state", state).Msg("ignoring invalid bits")
-		return
+		context.Warn().Msg("ignoring invalid bits")
+		state &= ValidKeyboardLedMasks
 	}
 
+	context = context.Uint8("old_state", u.keyboardState)
+
 	if u.keyboardState == state {
+		context.Trace().Msg("unchanged keyboardState")
 		return
 	}
-	u.log.Trace().Uint8("old", u.keyboardState).Uint8("new", state).Msg("keyboardState updated")
+
 	u.keyboardState = state
+	context.Trace().Msg("updated keyboardState")
 
-	if u.onKeyboardStateChange != nil {
-		(*u.onKeyboardStateChange)(getKeyboardState(state))
+	if cb := u.onKeyboardStateChange; cb != nil {
```

Comment:
HUGE change here... we need to ensure that when we trigger report-back transactions that they get emitted asynchronously so the host and device can get on with life while the network and/or client is tardy.